### PR TITLE
bpo-35089: Don't mention typing.io and typing.re

### PR DESCRIPTION
**Pull request by srittau**
https://bugs.python.org/issue35089

### CODE REVIEW

**Review by gvanrossum — APPROVED** [review]

**Comment by gvanrossum** [comment]
I suggest you add a NEWS entry.

File: Doc/library/typing.rst [hunk 1]
```diff
@@ -779,32 +779,25 @@ The module defines the following classes, functions and decorators:
...
+           TextIO
+           BinaryIO
 
-   Wrapper namespace for I/O stream types.
-
-   This defines the generic type ``IO[AnyStr]`` and subclasses ``TextIO``
-   and ``BinaryIO``, deriving from ``IO[str]`` and ``IO[bytes]``,
-   respectively. These represent the types of I/O streams such as returned by
+   Generic type ``IO[AnyStr]`` and subclasses ``TextIO(IO[str])``
+   and ``BinaryIO(IO[bytes])``
```
Comment (ilevkivskyi) [inline Doc/library/typing.rst:787]: I actually like the original formulation more. But this one is also fine.

Comment (gvanrossum) [reply Doc/library/typing.rst:787]: Maybe "and its subclasses".

Comment (srittau) [reply Doc/library/typing.rst:787]: The problem is that "This defined" is now wrong, but when removing the verb you have to reformulate. I first ended up with:

> The generic type `IO[AnyStr]` and subclasses `TextIO` and `BinaryIO`, deriving from `IO[str]` and `IO[bytes]`, respectively, represent the types of I/O streams such as returned by `open`.

 But I found this hard to understand.

Comment (ilevkivskyi) [reply Doc/library/typing.rst:787]: @srittau You can also try splitting the sentence in two parts. Anyway, this is not very important.

**Pull request merged**